helix-term/src/ui/completion.rs

```diff
@@ -33,8 +37,20 @@ impl menu::Item for CompletionItem {
     }
 
     fn format(&self, _data: &Self::Data) -> menu::Row {
+        let deprecated = if let Some(tags) = &self.tags {
+            tags.contains(&lsp::CompletionItemTag::DEPRECATED)
+        } else {
+            self.deprecated.unwrap_or_default()
+        };
```

Comment:
If we support more tags in the future it might be poasible that a server sets the dprecated field, doesn't send the deprecated tag but send other tags.
Lets be a bit more robust and check both: 
```suggestion
        let deprecated = self.deprecated.unwrap_or_default() || self.tags.map_or(false, |tags| tags.contains(&lsp::CompletionItemTag::DEPRECATED));
```

Reply:
Thanks for suggestion, I had to use
```
        let deprecated = self.deprecated.unwrap_or_default() || self.tags.as_ref().map_or(false, |tags| tags.contains(&lsp::CompletionItemTag::DEPRECATED));
```
(`self.tags.as_ref()` being the difference) instead so not applied directly from the commit suggestion.